spacy/pipeline/tok2vec.py

```diff
@@ -295,4 +314,7 @@ def forward(model: Tok2VecListener, inputs, is_train: bool):
         model.verify_inputs(inputs)
         return model._outputs, model._backprop
     else:
-        return [doc.tensor for doc in inputs], lambda dX: []
+        outputs = model._outputs
+        if outputs is None:
+            outputs = [model.ops.alloc2f(len(doc), model.get_dim("nO")) for doc in inputs]
```

Comment:
Should we unset the `_outputs` and `_backprop` here? It's kind of hard to know...Maybe it's good to allow multiple runs of the model? There's not really a great answer. Maybe we should rely on the batch ID rather than checking whether `_outputs` is None?

Reply:
Ah, yes, good idea to rely on `model._batch_id`, that also makes the code more clear. I don't really see why we should reset anything though? In theory the listener should be allowed to ask the results for the same batch ID twice?

Reply:
Yep, makes sense.